components/button/floating-buttons.js

```diff
@@ -0,0 +1,260 @@
+				border-top-color: var(--d2l-color-mica);
+				bottom: 0;
+				box-shadow: 0 -2px 4px rgba(86, 90, 92, .2);
+				left: 0;
+				position: fixed;
+				right: 0;
+				z-index: 999;
+			}
+			.d2l-floating-buttons-container > div {
+				padding: 0.75rem 0 0 0;
+				position: relative;
+			}
+			.d2l-floating-buttons-container > div {
+				padding: 0.75rem 0 0 0;
+				position: relative;
+			}
+			.d2l-floating-buttons-inner-container ::slotted(d2l-button),
+			.d2l-floating-buttons-inner-container ::slotted(button),
+			.d2l-floating-buttons-inner-container ::slotted(.d2l-button) {
+				margin-right: 0.75rem !important;
+				margin-bottom: 0.75rem !important;
+			}
+			:host-context([dir="rtl"]) .d2l-floating-buttons-inner-container ::slotted(d2l-button),
+			:host-context([dir="rtl"]) .d2l-floating-buttons-inner-container ::slotted(button),
+			:host-context([dir="rtl"]) .d2l-floating-buttons-inner-container ::slotted(.d2l-button) {
+				margin-left: 0.75rem !important;
+				margin-right: 0 !important;
+			}
+
+			@keyframes d2l-floating-buttons-animation {
+				0% {
+					border-color: transparent;
+					background-color: transparent;
+					transform: translate(0,10px);
+				}
+				100% {
+					border-top-color: var(--d2l-color-mica);
+					background-color: rgba(255, 255, 255, 0.88);
+					transform: translate(0,0);
+				}
+			}
+			@-webkit-keyframes d2l-floating-buttons-animation {
+				0% {
+					border-color: transparent;
+					background-color: transparent;
+					-webkit-transform: translate(0,10px);
+				}
+				100% {
+					border-top-color: var(--d2l-color-mica);
+					background-color: rgba(255, 255, 255, 0.88);
+					-webkit-transform: translate(0,0);
+				}
+			}
+			`
+		];
+	}
+
+	constructor() {
+		super();
+		this._container = null;
+		this._isRTL = false;
+		this._spacer = null;
+		this.minHeight = '500px';
+		this._containerClasses = {
+			'd2l-floating-buttons-container': true,
+			'd2l-floating-buttons-floating': false
+		};
+		this._innerContainerStyle = {
+			left: '',
+			right: '',
+			width: ''
+		};
+		this._buttonSpacerStyle = {
+			height: '',
+			display: ''
+		};
+	}
+
+	connectedCallback() {
+		super.connectedCallback();
+		window.addEventListener('resize', () => {
+			this._reposition();
+		});
+		window.addEventListener('scroll', () => {
+			this._reposition();
+		});
+	}
+
+	disconnectedCallback() {
+		super.disconnectedCallback();
+		window.removeEventListener('resize', () => {
+			this._reposition();
+		});
+		window.removeEventListener('scroll', () => {
+			this._reposition();
+		});
+	}
+
+	render() {
+		return html`
+			<div class=${classMap(this._containerClasses)} >
+				<div class="d2l-floating-buttons-inner-container"
+					style=${styleMap(this._innerContainerStyle)}>
+					<slot></slot>
+				</div>
+			</div>
+			<div class="d2l-floating-buttons-spacer"
+				style=${styleMap(this._buttonSpacerStyle)}>
+			</div>
+		`;
+	}
+
+	firstUpdated() {
+		this._reposition = this._reposition.bind(this);
```

Comment:
I think this bind should be moved to the constructor. The `bind` call returns a new function instance.  We want to ensure that our add/remove event listeners in the `connectedCallback` and `disconnectedCallback` are passing the same function instance, otherwise `removeEventListener` will not remove the handler that was added as mentioned above..